src/utils/geometry.rs

```diff
@@ -1338,6 +1350,26 @@ impl<N: Coordinate, Kind> Rectangle<N, Kind> {
     }
 }
 
+impl<N: Coordinate + Zero, Kind> Rectangle<N, Kind> {
+    /// Create a new [`Rectangle`] from its dimensions, with location zero
+    #[inline]
+    pub fn from_size(size: Size<N, Kind>) -> Self {
+        Rectangle {
+            loc: (Zero::zero(), Zero::zero()).into(),
+            size,
+        }
+    }
+
+    /// Create a new [`Rectangle`] with location and size zero
+    #[inline]
+    pub fn zero() -> Self {
+        Rectangle {
+            loc: (Zero::zero(), Zero::zero()).into(),
```

Comment:
i already added an associated `const ZERO` to the `Coordinate` trait in #1586, so you could drop the num-traits dependency and just use `N::ZERO` here.